### PR TITLE
FIX: skip sub directories when finding fonts on windows

### PR DESCRIPTION
**Pull request by Croadden**
add win32 fonts directory check

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by tacaswell** [comment]
Thank you for working on this @Croadden 

The test failures are real and caused by these changes because win32FontDirectory uses some windows-only modules.  This logic will have to be behind some platfrom gating.  I suspect that 👍🏻 

```python
if sys.platfrom == 'win32' and ...:

```

would be enough because `and` will short circuit and not evaluate the second expression if the first is `False`.

Could you please also give the PR title and the commit messages more descriptive text (like "FIX: skip sub directories when finding fonts on windows") so that someone who is coming across this issue / commit in the future will have are very quick suggestion as to the motivation for the change.  What files you changed and the actually changes is in the diff so the commit message is a chance for you to tell the future readers _why_ you made this change!

**Comment by Croadden** [comment]
Damn, was pretty sure I saw OS check :(
My bad, will fix. Sorry, I'm kinda new at github and whole this open source stuff

**Comment by Croadden** [comment]
@tacaswell, I was pretty sure I used propper commit naming... Also I'm not able to rename commits without force pushing and rebasing, so maybe lets leave it as is for now? Lesson learnt.

**Comment by Croadden** [comment]
Oh, found it! I'm using github desktop and I eddited _commit description_, but not _commit summary_. So that's why commit messages is _Update font_manager.py_ instead of _add win32 fonts directory check_ and _FIX: skip sub directories when finding fonts on windows_

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -196,11 +196,14 @@ def list_fonts(directory, extensions):
     recursively under the directory.
     """
     extensions = ["." + ext for ext in extensions]
-    return [os.path.join(dirpath, filename)
-            # os.walk ignores access errors, unlike Path.glob.
-            for dirpath, _, filenames in os.walk(directory)
-            for filename in filenames
-            if Path(filename).suffix.lower() in extensions]
+    if sys.platform == 'win32' and directory == win32FontDirectory():
+        return [os.path.join(directory, filename) for filename in os.listdir(directory) if os.path.isfile(filename)]
```
Comment (tacaswell) [inline lib/matplotlib/font_manager.py:200]: I am very confused why this is not failing the linter as too long of a line.....

Comment (Croadden) [reply lib/matplotlib/font_manager.py:200]: > I am very confused why this is not failing the linter as too long of a line.....

I'm running flake8 before each commit I make, so I think it's just ignored

Comment (Croadden) [reply lib/matplotlib/font_manager.py:200]: Oh, wrote previous comment before you replied to me :) 
So should I fix it or not?

**Comment by tacaswell** [comment]
> Oh, found it!

I'm glad you sorted it out!  The summary (e.g. the first line) is very helpful when skimming the history.

Unfortunately there is no way to update the commit message (which even though the UI shows it as two things is really just 1 block of text) is one of the values that git hashes to compute the SHA of the commit (which while annoying here, it means that someone can not change history without someone noticing!).  As changing the message changes the SHA (and hence history) you have to force-push to publish the new commits.  Again, git is trying to be helpful here and make sure that you do not accidentally delete history (think of the case where two people are working on the same branch.  If both fetch the upstream branch, do some work and commit.  The first person to push it works, the second person to push it fails (because if it did work it would effectively delete the commit from the first person).   From git's point of view it can not tell "I change a commit and want to update it" from "two people are trying to push completely different work" so `push` says no, but `git push --force-with-lease` says "trust me git, I know what I am doing!".


Sorry for the wall of text, hopefully it is more helpful than harmful.

**Comment by Croadden** [comment]
Your wall of the text help me so much, thanks! :)

But I made a lot of mistakes here. Sorry for being bad, my git experience is so low (was using SVN only for more than 5 years)
I fixes the issue with naming using rebase and everything seems fine in git and git gui (I have only 2 commits with proper naming here), but history seems broken here, at github. So what should I do? Maybe create another branch, delete this one and make another pull request?

File: lib/matplotlib/font_manager.py (hunk 1 above)
Comment (timhoffm) [inline lib/matplotlib/font_manager.py:200]:
```suggestion
        return [os.path.join(directory, filename)
                for filename in os.listdir(directory)
                if os.path.isfile(filename)]
```

**Comment by tacaswell** [comment]
@Croadden may I fix it and force-push to your branch?

**Comment by Croadden** [comment]
@tacaswell Sure, no problem, it's just simple reformat

**Comment by timhoffm** [comment]
@tacaswell as you go, you can include my suggestion above.

**Comment by tacaswell** [comment]
Done!

Steps I took:

 - use the GH ui to commit @timhoffm 's suggestion
 - checked out the branch locally
 - `git rebase -i` back to where this forked from master (I used magit in emacs to set this up) 
 - squashed everything down into one commit via the UI (could have maybe dropped duplicate commits, but squashing and fixing conflicts seemed like it would be faster)
 - fixed up a conflict as the rebase went
 - force pushed to your main branch: `git push -v --force-with-lease Croadden fix_win32_fonts:refs/heads/main` (again actually done through magit)
 - wrote this note

**Review by tacaswell — APPROVED** [review]

**Comment by Croadden** [comment]
Thanks!

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @Croadden! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**